File: game/src/main/java/com/scottlogic/hackathon/game/SpawnPoint.java
package com.scottlogic.hackathon.game;

import java.util.UUID;

/**
 * A spawn point in the game.
 */
public interface SpawnPoint {
    /**
     *
     * @return The unique id of the current spawn point.
     */
    UUID getId();

    /**
     *
     * @return The unique id of the owner of the current spawn point.
     */
    UUID getOwner();

    /**
     *
     * @return The position of the current spawn point.
     */
    Position getPosition();
}

File: python-contestant/src/game/spawn_point.py
from dataclasses import dataclass

from game.position import Position


@dataclass(frozen=True)
class SpawnPoint:
    """A spawn point in the game.

    Attributes:
        id (int): The unique ID of the current spawn point.
        position (Position): The position of the current spawn point.
        owner (int): The unique ID of the owner of the current spawn point.
    """
    id: int
    position: Position
    owner: int
